### PR TITLE
[ojelectronics] Adjust setpoint command handling

### LINKED ISSUE #16158: [ojelectronics] Error "Unable to set value 25" if setting temperatur value
## Expected Behavior
Setting a temperatur should work without error message

## Current Behavior
If a temperature value is set (via blockly), the temperature is not set to the item. Instead, there is a warning in the logs.

## Steps to Reproduce (for Bugs)
1. Define a simple rule
grafik
2. Error message in logs
`12:07:57.520 [WARN ] [ectronics.internal.ThermostatHandler] - Unable to set value 25`

## Your Environment
* Version used: OH4.1

### PR DESCRIPTION
**Pull request by EvilPingu**
# Description

Since version OH4.1 (maybe OH4) there are some problems with setting temperatur values. The reasons seems to be that sometimes the command is not a QuantityType<?> but a Number.

This PR fixes #16158

### CODE REVIEW

File: bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ private void updateManualSetpoint(ThermostatModel thermostat) {
     }
 
     private void updateManualSetpoint(Command command) {
-        if (command instanceof QuantityType<?> quantityCommand) {
+        if (command instanceof Number quantityCommand) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java:200]: Rather support the two options, Number and QuantityType.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java (hunk 1 above)
Comment (EvilPingu) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java:200]: I think QuantityType extends Number, so I think it should work, right?

[https://www.openhab.org/javadoc/latest/org/openhab/core/library/types/quantitytype](https://www.openhab.org/javadoc/latest/org/openhab/core/library/types/quantitytype)

Comment (lolodomo) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java:200]: In fact, I was thinking about `DecimalType `vs `QuantityType`. Yes, both are `Number`.
What is expected is that you can accept any temperature whatever its unit; that means values in ° Celsius or in ° Fahrenheit. So if you have a QuantityType as command value, you should first convert the value to the unit you finally expect (maybe Celsius, I don't know if this is the expected unit).
If you have no unit in the value you receive as command, you can consider a default unit of course.
Sorry, my comment was not very precise and not enough clear.

What is the unit expected for `getCurrentThermostat().manualModeSetpoint `?

Comment (EvilPingu) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java:200]: Now i got it. The target value is °C. So if the command is QuantityType i convert it to °C and if it's a decimal then i accept it always as °C. Is this the solution?

Comment (lolodomo) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java:200]: Exactly. You have many examples in other bindings implementing a set point channel.
Let me know if you need help to find such an example.

Comment (EvilPingu) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java]: I hope if found a better solution now. So please have a look again @lolodomo

Comment (lolodomo) [reply bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java]: That is OK

File: bundles/org.openhab.binding.ojelectronics/src/main/resources/OH-INF/i18n/ojelectronics_de.properties [hunk 2]
```diff
@@ -31,10 +31,10 @@ channel-type.ojelectronics.regulationMode.state.option.vacation = Urlaub
 channel-type.ojelectronics.regulationMode.state.option.frostProtection = Frostschutz
 channel-type.ojelectronics.regulationMode.state.option.boost = Boost
 channel-type.ojelectronics.regulationMode.state.option.eco = Spar
-channel-type.ojelectronics.comfortSetpoint.label = Komfort-Sollwert
+channel-type.ojelectronics.comfortSetpoint.label = Komfort-Zieltemperatur
```
[the remark this answers is not in the document]
Comment (EvilPingu) [reply bundles/org.openhab.binding.ojelectronics/src/main/resources/OH-INF/i18n/ojelectronics_de.properties]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by EvilPingu** [comment]
@lolodomo Thanks for your support

**Pull request merged**